sdks/go/pkg/beam/runners/prism/internal/environments.go

```diff
@@ -83,6 +96,33 @@ func runEnvironment(ctx context.Context, j *jobservices.Job, env string, wk *wor
 	}
 }
 
+func selectAnyOfEnv(ap *pipepb.AnyOfEnvironmentPayload) *pipepb.Environment {
+	// Prefer external, then process, then docker, unknown environments are 0.
+	ranks := map[string]int{
```

Comment:
This doesn't handle nested AnyOf. Do we care?

Reply:
I don't believe we do. We could, but I think we shouldn't.

I'd say it's to Beam's benefit to *not* encourage that sort of thing.